### PR TITLE
Update DateTime documentation

### PR DESCRIPTION
**Pull request by fchabouis**
@benwilson512 comments got 13 likes in a few hours, I thought it was worth updating the documentation to make the information easier to find !

https://elixirforum.com/t/why-is-datetime-module-lacking-min-and-max-functions/45164

### CODE REVIEW

File: lib/elixir/lib/calendar/datetime.ex [hunk 1]
```diff
@@ -13,7 +13,10 @@ defmodule DateTime do
 
   Remember, comparisons in Elixir using `==/2`, `>/2`, `</2` and friends
   are structural and based on the DateTime struct fields. For proper
-  comparison between datetimes, use the `compare/2` function.
+  comparison between datetimes, use the `compare/2` function. The
+  existence of the `compare/2` function in this module also allows
+  using `Enum.min/2` and `Enum.max/2` functions to get the minimum and
+  maximum datetime of an `Enum`.
```
Comment (josevalim) [inline lib/elixir/lib/calendar/datetime.ex:19]: Shall we include an example? :)

**Comment by pdgonzalez872** [comment]
Awesome!

Should the same happen in Date?

**Comment by josevalim** [comment]
It applies to all calendar types, so if we want to add the note and the example to all of them, it would certainly be welcome!

**Comment by fchabouis** [comment]
I added a similar explanation for `Date` and one example for each module.

File: lib/elixir/lib/calendar/date.ex [hunk 2]
```diff
@@ -31,7 +31,12 @@ defmodule Date do
 
   Comparisons in Elixir using `==/2`, `>/2`, `</2` and similar are structural
   and based on the `Date` struct fields. For proper comparison between
-  dates, use the `compare/2` function.
+  dates, use the `compare/2` function. The existence of the `compare/2`
+  function in this module also allows using `Enum.min/2` and `Enum.max/2`
+  functions to get the minimum and maximum date of an `Enum`. For example:
+
+      iex>  Enum.min([~D[2017-03-31], ~D[2017-04-01]], Date)
+      ~D[2017-03-31]
```
Comment (v0idpwn) [inline lib/elixir/lib/calendar/date.ex:39]: Perhaps this would be better in `compare/2` documentation instead of at moduledoc?

Comment (josevalim) [reply lib/elixir/lib/calendar/date.ex:39]: Moduledoc is better as introduction IMO but we can refer to mdocs from the regular doc.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by josevalim** [comment]
A PR for Time/NaiveDateTime is also welcome!

**Pull request merged**